### PR TITLE
Reformat Contributing (Reopened)

### PR DESCRIPTION
**Pull request by GNosii**
#### Description: 

Copied from the original PR at #6251:
> It's the time of my triyearly PR! This pull request reformats the `CONTRIBUTING.md` file by: 
> - Fixing grammar errors.
> - Simplifying rules, making them easier to understand.
> - Simplifying sections, removing redundant content.
> - Make use of new GFM features like footnotes.
> - Remove commented out section (pre-crowdin translation)
____
#### New Nodes/Commands/ConfigOptions: 

____
#### Relevant Towny Issue ticket:

____
- [ ] I have tested this pull request for defects on a server. 

By making this pull request, I represent that I have the right to waive copyright and related rights to my contribution, and agree that all copyright and related rights in my contributions are waived, and I acknowledge that the TownyAdvanced organization has the copyright to use and modify my contribution under the Towny [License](https://github.com/LlmDl/Towny/blob/master/LICENSE.md) for perpetuity.

### CODE REVIEW

File: .github/CONTRIBUTING.MD [hunk 1]
```diff
@@ -1,60 +1,61 @@
...
+Contributions are welcome via GitHub Pull Requests.
+You must fork this repository and then open it via your IDE.   
+You can always ask for help on your contribution in the [Towny Discord](https://discord.gg/gnpVs5m).    
+
+## Before contributing
+- Make sure to read the [Towny license](https://github.com/TownyAdvanced/Towny/blob/master/LICENSE.md).
+- By opening an pull request, you:
+    - represent that you have the right to waive copyright and related rights to your contributions
+    - agree that all copyright and related rights in your contributions will be waived
+    - acknowledge that the TownyAdvanced organization has the copyright to use and modify your contributions under the [Towny license](https://github.com/TownyAdvanced/Towny/blob/master/LICENSE.md) for perpetuity
```
Comment (LlmDl) [inline .github/CONTRIBUTING.MD:13]:
```suggestion
    - represent that you have the right to waive copyright and related rights to your contributions,
    - agree that all copyright and related rights in your contributions will be waived,
    - acknowledge that the TownyAdvanced organization has the copyright to use and modify your contributions under the [Towny license](https://github.com/TownyAdvanced/Towny/blob/master/LICENSE.md) for perpetuity.
```

File: .github/CONTRIBUTING.MD [hunk 2]
```diff
@@ -1,60 +1,61 @@
...
+
+If you do not agree with the terms of our license, do not contribute to the Towny project.
+
+# About pull requests
+> Pull requests will have their commits squashed upon merge, this also includes commit messages! Take this into account when commiting.
```
Comment (LlmDl) [inline .github/CONTRIBUTING.MD:18]:
```suggestion
> Pull requests will have their commits squashed upon merge, and your various commits' messages will be used in the squashed commit. Take this into account when making your commits.
```

File: .github/CONTRIBUTING.MD [hunk 3]
```diff
@@ -1,60 +1,61 @@
...
-Pull-Request your updated version. Please try to keep the title simple and include the version number:    
-e.g "German Translation v0.42"    
-If you are creating a new language file, feel-free to use either the english translation e.g "norwegian.yml" or the global placeholder e.g "es-mx"    
-If you are editing a language which contains the '(apostrophe) character such as French, please make sure you run it all through a YAML Parser,    
-You have to double up the '' when you want one ' to appear. It may help to do this anyways, as other characters may also require an escape sequence.-->
+3. Do not add plugin support directly into Towny **unless** it can benefit the project (such as economy/permission bridges).
+   - For Towny support, you should be using the [`TownyAPI`](https://github.com/TownyAdvanced/Towny/wiki/TownyAPI) instead.
+   - Even if you believe it can benefit the project, please ask the maintainer first.
+4. Do not embed dependencies directly into Towny.
+   - All dependencies should use the `provided` scope instead. This means that the dependency will be provided by the plugin JAR (e.g. Vault), or the server JAR (e.g. Adventure).
```
Comment (LlmDl) [inline .github/CONTRIBUTING.MD:48]:
```suggestion
   - All dependencies should use the `provided` scope instead. This means that the dependency will be provided by the plugin JAR (e.g. Vault), or the server JAR (e.g. Essentials, Citizens).
```

**Review by LlmDl — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.MD (hunk 3 above)
Comment (GNosii) [reply .github/CONTRIBUTING.MD:48]: Not sure what you mean here, by server JAR I'm talking about libraries included in paper, spigot, yk, adventure and gson for example

**Review by LlmDl — APPROVED** [review]

**Pull request merged**